content/docs/engine-dev/tools/tiaf/maintaining-tiaf/debugging-tiaf.md

```diff
@@ -77,47 +87,182 @@ In the log above, you can extract the following information about the AR run and
+```
+
+In your IDE, find the appropriate console front end project for the AR run (either `TestImpact.Frontend.Console.Native` or `TestImpact.Frontend.Console.Python`) and provide the arguments above. Now run the project with your debugger attached to find and fix the issue causing the AR failure.
+
+For this AR run, the cause of the crash was determined to be a bug due to an edge case where a test target had previously been opted-in to the TIAF (whereupon its coverage data was stored in the `historic_data.json` file used by the run), only for that test target to have been subsequently opted-out at a later date, causing a `nullptr` dereference that was fixed in commit `98dbed226cdae5afa5da25fe4bf84ae6760e4ac4`.
+
+{{< tip >}}
+You will save a lot of time by reproducing issues with the TIAF itself locally rather than attempting to debug them in AR with the limited tooling available.
+{{< /tip >}}
+
+### Running failing tests
+
+If the cause of the AR failure is a specific test target failing or crashing, find the offending test target in your IDE and run it with your debugger attached.
+
+To manually launch a given test target from the command line as per the method used by TIAF (for example, to inspect the test run and coverage data), open up the `historic_data.json` in the `Persistent` directory and locate the test target's `command_args` field in that file. For example, the (truncated) historic data for the local run in the first step shows the command line arguments needed to invoke the `Gem_Metastream.Tests` test target from your terminal:
+
+```json
+"last_commit_hash":"4b65521188a2a0beb399eca7b692216d2fe3d209",
+   "historic_sequences":{
+      "4b65521188a2a0beb399eca7b692216d2fe3d209":null
+   },
+   "previous_test_runs":[
+      {
+         "name":"Gem_Metastream.Tests",
+         "command_args":"\"C:\\dev\\OpenCppCoverage\\x64\\Release\\OpenCppCoverage.exe\" --coverage_level source --export_type cobertura:\"C:\\dev\\o3de\\build\\windows_vs2019\\bin\\TestImpactFramework\\debug\\Temp\\Native\\Coverage\\Metastream.Tests.xml\" --modules \"C:\\dev\\o3de\\build\\windows_vs2019\\bin\\debug\" --excluded_modules \"C:\\dev\\o3de\\build\\windows_vs2019\\bin\\debug\\AzTestRunner.exe\" --sources \"C:\\dev\\o3de\" -- \"C:\\dev\\o3de\\build\\windows_vs2019\\bin\\debug\\AzTestRunner.exe\" \"C:\\dev\\o3de\\build\\windows_vs2019\\bin\\debug\\Metastream.Tests.dll\" AzRunUnitTests --gtest_output=xml:\"C:\\dev\\o3de\\build\\windows_vs2019\\Testing\\Gtest\\Gem_Metastream.Tests.xml\" ",
+         "start_time":2257721140,
+         "end_time":2258136771,
+         "duration":415631,
+         "result":"all_tests_pass",
+         "num_passing_tests":1,
+         "num_failing_tests":0,
+         "num_disabled_tests":0,
+         "tests":[
+            {
+               "name":"MetastreamTest.ServerStartupShutdownTest_FT",
+               "result":"passed"
+            }
+         ]
+      },
+      // rest of data removed for clarity
+```
```

Comment:
````suggestion
    ```json
    "last_commit_hash":"4b65521188a2a0beb399eca7b692216d2fe3d209",
    "historic_sequences":{
        "4b65521188a2a0beb399eca7b692216d2fe3d209":null
    },
    "previous_test_runs":[
        {
            "name":"Gem_Metastream.Tests",
            "command_args":"\"C:\\dev\\OpenCppCoverage\\x64\\Release\\OpenCppCoverage.exe\" --coverage_level source --export_type cobertura:\"C:\\dev\\o3de\\build\\windows_vs2019\\bin\\TestImpactFramework\\debug\\Temp\\Native\\Coverage\\Metastream.Tests.xml\" --modules \"C:\\dev\\o3de\\build\\windows_vs2019\\bin\\debug\" --excluded_modules \"C:\\dev\\o3de\\build\\windows_vs2019\\bin\\debug\\AzTestRunner.exe\" --sources \"C:\\dev\\o3de\" -- \"C:\\dev\\o3de\\build\\windows_vs2019\\bin\\debug\\AzTestRunner.exe\" \"C:\\dev\\o3de\\build\\windows_vs2019\\bin\\debug\\Metastream.Tests.dll\" AzRunUnitTests --gtest_output=xml:\"C:\\dev\\o3de\\build\\windows_vs2019\\Testing\\Gtest\\Gem_Metastream.Tests.xml\" ",
            "start_time":2257721140,
            "end_time":2258136771,
            "duration":415631,
            "result":"all_tests_pass",
            "num_passing_tests":1,
            "num_failing_tests":0,
            "num_disabled_tests":0,
            "tests":[
                {
                "name":"MetastreamTest.ServerStartupShutdownTest_FT",
                "result":"passed"
                }
            ]
        }
    
        // rest of data removed for clarity
    ]
    ```
````